src/components/ui4/CurrencyViewUi4.tsx

```diff
@@ -0,0 +1,80 @@
+import { EdgeCurrencyWallet, EdgeToken } from 'edge-core-js'
+import * as React from 'react'
+import { Text } from 'react-native'
+
+import { SPECIAL_CURRENCY_INFO } from '../../constants/WalletAndCurrencyConstants'
+import { useWalletBalance } from '../../hooks/useWalletBalance'
+import { useWalletName } from '../../hooks/useWalletName'
+import { lstrings } from '../../locales/strings'
+import { useSelector } from '../../types/reactRedux'
+import { IconDataRow } from '../data/row/IconDataRow'
+import { CryptoIcon } from '../icons/CryptoIcon'
+import { useTheme } from '../services/ThemeContext'
+import { CryptoText } from '../text/CryptoText'
+import { FiatText } from '../text/FiatText'
+import { TickerText } from '../text/TickerText'
+
+interface Props {
+  marginRem?: number[] | number
+  nativeAmount?: string
+  showRate?: boolean
+  token?: EdgeToken
+  tokenId?: string
+  wallet: EdgeCurrencyWallet
+}
+
+/**
+ * A view representing the data from a wallet, used for rows, cards, etc.
+ */
+const CurrencyViewUi4Component = (props: Props) => {
+  const { nativeAmount, showRate = false, token, tokenId, wallet } = props
+  const { pluginId } = wallet.currencyInfo
+  const { showTokenNames = false } = SPECIAL_CURRENCY_INFO[pluginId] ?? {}
+  const theme = useTheme()
+
+  // Currency code for display:
+  const allTokens = wallet.currencyConfig.allTokens
+  const tokenFromId = token != null ? token : tokenId == null ? null : allTokens[tokenId]
+  const { currencyCode } = tokenFromId == null ? wallet.currencyInfo : tokenFromId
+
+  // Wallet name for display:
+  let name: React.ReactNode = useWalletName(wallet)
+  const compromised = useSelector(state => {
+    const { modalShown = 0 } = state.ui?.settings?.securityCheckedWallets?.[wallet.id] ?? {}
+    return modalShown > 0
+  })
+  if (compromised) {
+    name = (
+      <>
+        <Text style={{ color: theme.warningText }}>{lstrings.compromised_key_label}</Text> {name}
+      </>
+    )
+  }
+
+  // Balance stuff:
+  const showBalance = useSelector(state => state.ui.settings.isAccountBalanceVisible)
+  const balance = useWalletBalance(wallet, tokenId)
+  const icon = <CryptoIcon sizeRem={2} tokenId={tokenId} walletId={wallet.id} />
+  const tickerText = showRate && wallet != null ? <TickerText wallet={wallet} tokenId={tokenId} /> : null
+  const cryptoText = showBalance ? <CryptoText wallet={wallet} tokenId={tokenId} nativeAmount={nativeAmount ?? balance} withSymbol /> : null
+  const fiatText = showBalance ? <FiatText nativeCryptoAmount={nativeAmount ?? balance} tokenId={tokenId} wallet={wallet} /> : null
```

Comment:
👍 This is really clean & nice. It looks like all your hard work is really starting to pay off, now that we have these nice re-usable components for showing amounts in various ways. I am really impressed, Jon!